searchcore/src/tests/proton/documentdb/lid_space_compaction/lid_space_common.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "lid_space_common.h"
+
+MyScanIterator::MyScanIterator(const LidVector &lids, bool bucketIdEqualLid)
+    : _lids(lids),
+      _itr(_lids.begin()),
+      _validItr(true),
+      _bucketIdEqualLid(bucketIdEqualLid)
+{}
+MyScanIterator::~MyScanIterator() = default;
+
+bool
+MyScanIterator::valid() const {
+    return _validItr;
+}
+search::DocumentMetaData MyScanIterator::next(uint32_t compactLidLimit, bool retry) {
+    if (!retry && _itr != _lids.begin()) {
+        ++_itr;
+    }
+    for (; _itr != _lids.end() && (*_itr) <= compactLidLimit; ++_itr) {}
+    if (_itr != _lids.end()) {
+        uint32_t lid = *_itr;
+        if (lid > compactLidLimit) {
+            return search::DocumentMetaData(lid, TIMESTAMP_1, createBucketId(lid), GID_1);
+        }
+    } else {
+        _validItr = false;
+    }
+    return search::DocumentMetaData();
+}
+search::DocumentMetaData MyScanIterator::getMetaData(uint32_t lid) const {
```

Comment:
Consider newline between functions. Applies for a lot of functions further down as well.

Reply:
Fixed